packages/lit-dev-tools-esm/src/build-unversioned-docs.ts

```diff
@@ -0,0 +1,154 @@
+  'site',
+  'docs',
+  'unversioned'
+);
+
+/**
+ * This script builds our unversioned latest documentation for lit.dev. It
+ * locates the current latestVersion of Lit to show from `site.json`, then
+ * copies that documentation from the versioned subdirectory into an unversioned
+ * subdirectory.
+ *
+ * The following transforms are then applied on the files:
+ *   - Permalink frontmatter is added to strip the version from the URL. E.g.,
+ *     /docs/v2/* becomes /docs/*.
+ *   - Versioned cross links are detected and made unversioned.
+ */
+const buildAndTransformUnverionedDocs = async () => {
+  const walk = async (dirPath: string): Promise<void> => {
+    const entries = await fsPromise.readdir(dirPath, {withFileTypes: true});
+    await Promise.all(
+      entries.map(async (entry) => {
+        const childPath = pathlib.join(dirPath, entry.name);
+        if (entry.isDirectory()) {
+          return walk(childPath);
+        } else {
+          transformFile(childPath);
+        }
+      })
+    );
+  };
+
+  console.log('Starting build-unversioned-docs.js script');
+  await walk(LATEST_VERSION_CONTENT);
+  console.log('Completed build-unversioned-docs.js script');
+};
+
+/**
+ * Transform the given versioned content file and copy it to the unversioned
+ * directory.
+ */
+function transformFile(path: string) {
+  let fileContents = fs.readFileSync(path, {encoding: 'utf8'});
+  const relativeChildPath = pathlib.relative(LATEST_VERSION_CONTENT, path);
+  const ext = pathlib.extname(relativeChildPath);
+  let unversionedLocation = pathlib.join(
+    UNVERSIONED_VERSION_LOCATION,
+    relativeChildPath
+  );
+
+  if (ext === '.md') {
+    const fileName = pathlib.basename(unversionedLocation, '.md');
+    const [frontMatterData, restOfFile] = getFrontMatterData(fileContents);
+    const existingPermalink = frontMatterData.findIndex((val) =>
+      val.includes('permalink:')
+    );
+    if (existingPermalink !== -1) {
+      throw new Error(
+        'Unhandled case: Handle this by transforming the permalink here.'
+      );
+    }
+    if (fileName === 'index') {
+      frontMatterData.push(
+        `permalink: docs/${relativeChildPath.slice(0, -3)}.html`
+      );
+    } else {
+      frontMatterData.push(
+        `permalink: docs/${relativeChildPath.slice(0, -3)}/index.html`
+      );
+    }
+
+    fileContents = writeFrontMatter(frontMatterData) + restOfFile;
+  } else if (ext === '.json') {
+    if (
+      pathlib.basename(unversionedLocation, '.json') === SITE_LATEST_VERSION
+    ) {
+      unversionedLocation = pathlib.join(
+        pathlib.dirname(unversionedLocation),
+        'unversioned.json'
+      );
+      fileContents = JSON.stringify({
+        collection: 'docs-unversioned',
+        latestVersion: SITE_LATEST_VERSION,
+      });
+    }
+  } else if (ext === '.html') {
+    if (pathlib.basename(unversionedLocation, '.html') === 'api') {
+      const [frontMatterData, _] = getFrontMatterData(fileContents);
```

Comment:
unused trailing value from array destructuring can be dropped, though maybe you left that intentionally there to signal it is intentionally not used, please ignore this comment if so.
```suggestion
      const [frontMatterData] = getFrontMatterData(fileContents);
```

but do .html files never have content following front matter?

Reply:
Great question! This is scoped to `api.html` which only contains frontmatter.

See: https://github.com/lit/lit.dev/blob/main/packages/lit-dev-content/site/docs/api/api.html

Reply:
Added some more docs to cover this case